components/nexus-core/src/main/java/org/sonatype/nexus/web/internal/CookieFilter.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2008-2015 Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.web.internal;
+
+import java.util.Collection;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletResponse;
+
+import com.google.common.net.HttpHeaders;
+import org.apache.shiro.web.servlet.AdviceFilter;
+
+/**
+ * Cookie munging filter.
+ *
+ * @since 2.11.2
+ */
+@Named
+@Singleton
+public class CookieFilter
+    extends AdviceFilter
+{
+  private static final String SECURE_FLAG = "; Secure";
+
+  private static final String SET_COOKIE = HttpHeaders.SET_COOKIE;
```

Comment:
Using HttpHeaders and referencing in a private member was an afterthought since I wasn't sure if importing that class at all was worth it.

I guess I will depend on HttpHeaders directly as no-one seemed to complain about that specifically.